### PR TITLE
[scheduler] Parallelize ci.yaml scheduling requests

### PR DESCRIPTION
**Pull request by CaseyHillers**
The framework presubmit check takes about ~1 minute to schedule everything. The culprit appears to be running everything synchronously.

* Use `Future.wait` to process the various requests in parallel
* Small refactor to move the core logic into a re-usable function (_scheduleTryBuild)

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: app_dart/lib/src/service/luci_build_service.dart [hunk 1]
```diff
@@ -180,53 +179,18 @@ class LuciBuildService {
...
+      futures.add(
+        _scheduleTryBuild(
+          target: target,
+          sha: sha,
+          pullRequest: pullRequest,
+          cipdVersion: cipdVersion,
         ),
       );
     }
+    final List<Request> requests = await Future.wait(futures);
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:193]: I doubt this will help since there is no async operation in `_scheduleTryBuild` at all

Comment (CaseyHillers) [reply app_dart/lib/src/service/luci_build_service.dart:193]: githubChecksUtil.createCheckRun is async, and seems to take about 1 second

Comment (keyonghan) [reply app_dart/lib/src/service/luci_build_service.dart:193]: Yeah, you are right.

One concern is the `second rate limit`. With this change, I expect the API will be called more frequently within a short time and will increase the chance to hit the second rate limit.

IIUC, this PR aims to reduce the show up latency of these check runs, but doesn't seem to help with the LUCI builds scheduling/execution time.

Comment (CaseyHillers) [reply app_dart/lib/src/service/luci_build_service.dart:193]: For the secondary rate limit, my understanding is that is only an issue on authentication. This is for repeat calls to an existing client. Or is there some other logic happening under the hood now?

For the LUCI scheduling, that is published to the scheduler-reqeusts topic which should take ~10ms. We could parallelize it if there's interest, but I estimate it's <1% of the framework time.

Comment (CaseyHillers) [reply app_dart/lib/src/service/luci_build_service.dart:193]: Thanks for the link, I'll close this. However, Cocoon isn't implementing this best practice today, so I filed https://github.com/flutter/flutter/issues/142816 for tracking.

**Pull request closed without merging**